src/client.js

```diff
@@ -46,9 +46,11 @@ export function streamEvents(system, callback) {
 
 export function getInfo(system) {
     return new Promise((resolve, reject) => {
+        const timeout = setTimeout(() => reject(new Error("timeout")), 5000);
```

Comment:
If the answer comes back eventually, this won't mess up the protocol and our code? AFAIR calling resolve() after reject() already got  called seems safe. But wrt. the protocol, if getInfo() fails then we don't do any other API call. `</thinking_aloud>`. So should be fine.